table/block_based_table_builder.cc

```diff
@@ -650,16 +650,26 @@ Status BlockBasedTableBuilder::Finish() {
 
   BlockHandle filter_block_handle, metaindex_block_handle, index_block_handle,
       compression_dict_block_handle, range_del_block_handle;
-  // Write filter block
+  // START Write filter block
+  std::vector<Slice> filter_blocks;
   if (ok() && r->filter_builder != nullptr) {
     Status s = Status::Incomplete();
     while (s.IsIncomplete()) {
       Slice filter_content = r->filter_builder->Finish(filter_block_handle, &s);
       assert(s.ok() || s.IsIncomplete());
       r->props.filter_size += filter_content.size();
       WriteRawBlock(filter_content, kNoCompression, &filter_block_handle);
+      filter_blocks.push_back(filter_content);
     }
   }
+  // END write filter block
+
+  // Locally replicate filter block(s) again.
+  BlockHandle filter_block_handle2;
+  for (std::vector<Slice>::iterator it = filter_blocks.begin();
+        it != filter_blocks.end(); it++) {
+    WriteRawBlock(*it, kNoCompression, &filter_block_handle2);
```

Comment:
Can you guard it as an option?

Reply:
Guarded everything with BlockBasedTableOption::double_metadata